src/components/CropSelector/CropCalendarView/CropCalendarView.js

```diff
@@ -43,85 +39,10 @@ const CropCalendarViewComponent = (props) => {
   const { activeCropData } = props;
   const { state, dispatch } = useContext(Context);
   const [legendModal, setLegendModal] = useState(false);
-  const selectedBtns = state.selectedCrops.map((crop) => {
-    return crop.id;
-  });
 
   const handleLegendModal = () => {
     setLegendModal(!legendModal);
   };
-
-  const hasGoalRatingTwoOrLess = (crop = []) => {
-    const { selectedGoals } = state;
-
-    return crop.inactive || selectedGoals.every((rating) => crop.fields[rating] <= 2);
-  };
-
-  const addCropToBasket = (cropId, cropName, btnId, cropData) => {
-    let selectedCrops = {};
-    let cropArray = [];
-    selectedCrops['id'] = cropId;
-    selectedCrops['cropName'] = cropName;
-    selectedCrops['btnId'] = btnId;
-    selectedCrops['data'] = cropData;
-    cropArray = selectedCrops;
-
-    // // check if crop id exists inside state, if yes then remove it
-    if (state.selectedCrops.length > 0) {
-      // DONE: Remove crop from basket
-      let removeIndex = state.selectedCrops
-        .map(function (item) {
-          return item.btnId;
-        })
-        .indexOf(`${btnId}`);
-      if (removeIndex === -1) {
-        // element not in array
-        dispatch({
-          type: 'SELECTED_CROPS_MODIFIER',
-          data: {
-            selectedCrops: [...state.selectedCrops, selectedCrops],
-            snackOpen: true,
-            snackMessage: `${cropName} Added`,
-          },
-        });
-      } else {
-        let selectedCropsCopy = state.selectedCrops;
-        selectedCropsCopy.splice(removeIndex, 1);
-        dispatch({
-          type: 'SELECTED_CROPS_MODIFIER',
-          data: {
-            selectedCrops: selectedCropsCopy,
-            snackOpen: true,
-            snackMessage: `${cropName} Removed`,
-          },
-        });
-      }
-    } else {
-      // DONE: add the selected crop to state and change the state, show snackbar
-
-      dispatch({
-        type: 'SELECTED_CROPS_MODIFIER',
-        data: {
-          selectedCrops: [cropArray],
-          snackOpen: true,
-          snackMessage: `${cropName} Added`,
-        },
-      });
-    }
-  };
-
-  const getAverageGoalRating = (selectedGoals, crop) => {
-    // get goal rating for each crop and calculate+render rating
-    let goalRating = 0;
-    selectedGoals.forEach((goal) => {
-      if (crop.fields[goal]) {
-        goalRating += crop.fields[goal];
-      }
-    });
-
-    return getRating(goalRating / selectedGoals.length);
-  };
-
   const [activeGrowthPeriodState, setActiveGrowthPeriodState] = useState(state.activeGrowthPeriod);
 
   useEffect(() => {
```

Comment:
useEffects should be very thin.  Some functionality cannot be avoided,  but i can see some of this logic could be broken out into is own function and called in the use effect.  this will make the use effect easier to read and understand the flow.  also this will open us up to re useable code. 

ex.  the code block below is very similar in 4 places.  the same dispatch and type... i think there are more dispatches too.  maybe we could have a function for this call specifically.. other ideas... if we have a wide variety of different dispatch calls we could just write one function and pass in the differences. 
dispatch({
      type: 'UPDATE_ACTIVE_CROP_DATA',
      data: {
        value: activeCropDataShadow,
      },
    });